hydroflow_lang/src/graph/ops/dest_sink_serde_tcp.rs

```diff
@@ -0,0 +1,79 @@
+use super::{
+    OperatorConstraints, OperatorWriteOutput, WriteContextArgs, WriteIteratorArgs, RANGE_0, RANGE_1,
+};
+
+use quote::quote_spanned;
+
+/// > Arguments: A [serializing async `Sink`](https://docs.rs/futures/latest/futures/sink/trait.Sink.html).
+///
+/// Consumes (payload, addr) pairs by serializing the payload and sending the resulting pair to an [async `Sink`](https://docs.rs/futures/latest/futures/sink/trait.Sink.html).
+///
+/// Note this operator must be used within a Tokio runtime.
+/// ```rustbook
+/// async fn serde_out() {
+///     let addr = hydroflow::util::ipv4_resolve("localhost:9000".into()).unwrap();
+///     let (outbound, inbound, _) = hydroflow::util::tcp_bytes(addr).await;
+///     let remote = hydroflow::util::ipv4_resolve("localhost:9001".into()).unwrap();
+///     let mut flow = hydroflow::hydroflow_syntax! {
+///         source_iter(vec![("hello".to_string(), 1), ("world".to_string(), 2)])
+///             -> map (|m| (m, remote)) -> dest_sink_serde_tcp(outbound);
+///     };
+///     flow.run_available();
+/// }
+/// ```
+#[hydroflow_internalmacro::operator_docgen]
+pub const DEST_SINK_SERDE_TCP: OperatorConstraints = OperatorConstraints {
+    name: "dest_sink_serde_tcp",
+    hard_range_inn: RANGE_1,
+    soft_range_inn: RANGE_1,
+    hard_range_out: RANGE_0,
+    soft_range_out: RANGE_0,
+    ports_inn: None,
+    ports_out: None,
+    num_args: 1,
+    input_delaytype_fn: &|_| None,
+    write_fn: &(|wc @ &WriteContextArgs { root, op_span, .. },
+                 &WriteIteratorArgs {
+                     ident, arguments, ..
+                 },
+                 _| {
+        let sink_arg = &arguments[0];
+
+        let send_ident = wc.make_ident("item_send");
+        let recv_ident = wc.make_ident("item_recv");
+
+        let write_prologue = quote_spanned! {op_span=>
+            let (#send_ident, #recv_ident) = #root::tokio::sync::mpsc::unbounded_channel();
+            df
+                .spawn_task(async move {
+                    use #root::futures::sink::SinkExt;
+
+                    let mut recv = #recv_ident;
+                    let mut sink = #sink_arg;
+                    while let Some(payload) = recv.recv().await {
+                        let item = #root::util::serialize_to_bytes(payload);
```

Comment:
Is there a reason you didn't go with?
```rust
-> map(|payload| serialize_to_bytes(payload)) -> dest_sink(outbound)
```

Reply:
Lack of experience navigating hydroflow :)